back/src/domains/marketing/adapters/establishmentMarketingGateway/BrevoEstablishmentMarketingGateway.ts

```diff
@@ -0,0 +1,358 @@
+import { isAxiosError } from "axios";
+import Bottleneck from "bottleneck";
+import { Email } from "shared";
+import { HttpClient } from "shared-routes";
+import { BrevoHeaders } from "../../../../utils/apiBrevoUrl";
+import {
+  EstablishmentMarketingGateway,
+  EstablishmentMarketingGatewayDto,
+  EstablishmentMarketingSearchableBy,
+} from "../../ports/EstablishmentMarketingGateway";
+import { BrevoContactRoutes } from "./BrevoContactRoute";
+import {
+  CreateContactBody,
+  GetContactInfoResponseBody,
+  TypePublic,
+} from "./BrevoContactRoute.dto";
+
+type BrevoEstablishmentMarketingGatewayParams = {
+  httpClient: HttpClient<BrevoContactRoutes>;
+  apiKey: string;
+  establishmentContactListId: number;
+};
+
+const brevoMaxContactRequestsPerSeconds = 5;
+const ONE_SECOND_MS = 1_000;
+
+export type BrevoContact =
+  | {
+      contact: EstablishmentMarketingGatewayDto;
+      lists: number[];
+    }
+  | undefined;
+
```

Comment:
On peut faire : 
```suggestion
export type BrevoContact = {
      contact: EstablishmentMarketingGatewayDto;
      lists: number[];
    }
```

Et préciser explicitement les undefined ? Je met un exemple plus bas